custom_components/cardata/sensor.py

```diff
@@ -39,11 +49,41 @@ async def async_added_to_hass(self) -> None:
             if last_state and last_state.state not in ("unknown", "unavailable"):
                 self._attr_native_value = last_state.state
                 unit = last_state.attributes.get("unit_of_measurement")
+
                 if unit is not None:
+                    # HomeAssistant expects liters with capital L. 
+                    if unit == "l":
+                        unit = "L"
+
                     self._attr_native_unit_of_measurement = unit
-                    # If unit is a length/distance type, enable conversion
+                    # If unit is recognized, set _attr_device_class to enable conversion
                     if unit in {u.value for u in UnitOfLength}:
                         self._attr_device_class = SensorDeviceClass.DISTANCE # Enables km/mi, m/ft, etc., conversion
+
+                    if unit in {u.value for u in UnitOfPressure}:
```

Comment:
I am not a python expert, but would it make sense to make these "elif" to avoid repeated compares?